src/messageComposer/middleware/textComposer/commandStringExtraction.ts

```diff
@@ -0,0 +1,55 @@
+import type { Middleware } from '../../../middleware';
+import type { TextComposerMiddlewareExecutorState } from './TextComposerMiddlewareExecutor';
+import { stripTriggerFromText } from './textMiddlewareUtils';
+import type { CommandSuggestion } from './types';
+
+export type ApplyCommandSettingsMiddleware = Middleware<
+  TextComposerMiddlewareExecutorState<CommandSuggestion>,
+  'onChange' | 'onSuggestionItemSelect'
+>;
+
+export const createCommandStringExtractionMiddleware =
+  (): ApplyCommandSettingsMiddleware => ({
+    handlers: {
+      onChange: ({ complete, forward, state }) => {
+        const { command } = state;
+
+        if (!command) {
```

Comment:
We should also forward if the command in the composer state equals this command.  That way we can also avoid setting incorrect selection:

Here the text is the actual argument to the command (for example `@`), so we cannot subtract the command length from a text that has 1 char for example.
```
        end: state.selection.end - triggerWithCommand.length,
         start: state.selection.start - triggerWithCommand.length,
```